doc/api/stream.md

```diff
@@ -1909,6 +1907,28 @@ async function run() {
 run().catch(console.error);
 ```
 
+Remember to handle the `signal` argument passed into the async generator.
+Especially in the case where the async generator is the source for the
+pipeline (i.e. first argument) or the pipeline will never complete.
+
+```js
+const { pipeline } = require('stream/promises');
+const fs = require('fs');
+
+async function run() {
+  await pipeline(
+    async function * (signal) {
+      await someLongRunningfn({ signal });
```

Comment:
The signal should be checked to see if it has already been tripped before kicking off the async await.

Reply:
The inside won’t run if the signal has been tripped.